spring-cloud-alibaba-tests/nacos-tests/nacos-config-test/src/test/java/com/alibaba/cloud/tests/nacos/config/NacosConfigRefreshTest.java

```diff
@@ -16,132 +16,84 @@
-import com.alibaba.cloud.testsupport.HasDockerAndItEnabled;
+import java.util.Properties;
+
+import com.alibaba.cloud.testsupport.SpringCloudAlibaba;
+import com.alibaba.cloud.testsupport.TestExtend;
 import com.alibaba.cloud.testsupport.Tester;
+import com.alibaba.nacos.api.PropertyKeyConst;
+import com.alibaba.nacos.api.config.ConfigFactory;
 import com.alibaba.nacos.api.config.ConfigService;
 import com.alibaba.nacos.api.exception.NacosException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.testcontainers.containers.GenericContainer;
-
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.context.event.ApplicationReadyEvent;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.context.annotation.Import;
-import org.springframework.context.event.EventListener;
-import org.springframework.core.Ordered;
-import org.springframework.core.annotation.Order;
+import org.mockito.Mock;
 
-import static com.alibaba.cloud.tests.nacos.config.NacosConfigRefreshTest.PushConfigConfiguration;
+import static com.alibaba.cloud.testsupport.Constant.TIME_OUT;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
 
 /**
  *
  * Test function: Nacos config refresh.
  *
  * @author freeman
  */
-@HasDockerAndItEnabled
-@SpringBootTest(classes = NacosConfigTestApplication.class, webEnvironment = RANDOM_PORT, properties = {
-		"spring.profiles.active=nacos-config-refresh"
-})
-@Import(PushConfigConfiguration.class)
+// @HasDockerAndItEnabled
+@SpringCloudAlibaba(composeFiles = "docker/nacos-compose-test.yml", serviceName = "nacos-standalone")
+@TestExtend(time = 3 * TIME_OUT)
 public class NacosConfigRefreshTest {
 
-	static GenericContainer nacos = ContainerStarter.startNacos();
+	/**
+	 * nacos upload conf file.
+	 */
+	public static final String YAML_CONTENT = "configdata:\n" + "  user:\n"
+			+ "    age: 22\n" + "    name: freeman1123\n" + "    map:\n"
+			+ "      hobbies:\n" + "        - art\n" + "        - programming\n"
+			+ "        - movie\n" + "      intro: Hello, I'm freeman\n"
+			+ "      extra: yo~\n" + "    users:\n" + "      - name: dad\n"
+			+ "        age: 20\n" + "      - name: mom\n" + "        age: 18";
+
+	@Mock
+	protected ConfigService service1;
 
-	private static final String serverAddr;
+	@BeforeAll
+	public static void setUp() {
 
-	static {
-		serverAddr = "localhost:" + nacos.getMappedPort(8848);
-		System.setProperty("spring.cloud.nacos.config.server-addr", serverAddr);
 	}
 
-	@Autowired
-	private NacosConfigManager nacosConfigManager;
-	@Autowired
-	private UserProperties userProperties;
+	@BeforeEach
+	public void prepare() throws NacosException {
+		Properties nacosSettings = new Properties();
+		String serverIp8 = "127.0.0.1:8848";
```

Comment:
Variable naming of `serverIp8` is irregular.

Reply:
solved